File: subprojects/frontend/react-app/src/common/domain/Page.js
export default class Page {

  size;
  totalElements;
  totalPages;
  number;
  list;

  constructor(result, resourceListName, itemFactory) {
    console.log('result', result);
    this.size = result.page.size;
    this.totalElements = result.page.totalElements;
    this.totalPages = result.page.totalPages;
    this.number = result.page.number;
    this.list = [];

    if (result._embedded && result._embedded[resourceListName]) {
      for (const item of result._embedded[resourceListName]) {
        this.list.push(itemFactory(item));
      }
    }
  }
}

File: subprojects/backend/src/test/java/com/example/demo/support/Page.java
package com.example.demo.support;

import lombok.Data;

public class Page {

    private Long size;
    private Long totalElements;
    private Long totalPages;
    private Long number;

    public Long getSize() {
        return size;
    }

    public void setSize(Long size) {
        this.size = size;
    }

    public Long getTotalElements() {
        return totalElements;
    }

    public void setTotalElements(Long totalElements) {
        this.totalElements = totalElements;
    }

    public Long getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Long totalPages) {
        this.totalPages = totalPages;
    }

    public Long getNumber() {
        return number;
    }

    public void setNumber(Long number) {
        this.number = number;
    }
}
